### PR TITLE
[MON-1838] Upgrade Prometheus-Operator to 0.52

### PR DESCRIPTION
**Pull request by raptorsun**
* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

This PR upgrades Prometheus-Operator to version 0.52.
This is necessary dependency upgrade for PR #1467 .

### CODE REVIEW

**Review by PhilipGough — APPROVED** [review]

**Comment by simonpasquier** [comment]
/hold

I don't see a corresponding PR for the v0.52.0 bump in https://github.com/openshift/prometheus-operator. Am I missing something?

**Comment by raptorsun** [comment]
> /hold
> 
> I don't see a corresponding PR for the v0.52.0 bump in https://github.com/openshift/prometheus-operator. Am I missing something?

I'm going to make a PR of version bump for https://github.com/openshift/prometheus-operator

**Comment by juzhao** [comment]
@raptorsun prometheus-operator  version is still 0.51.2, although the resource labels is 0.52.0, waiting for https://github.com/openshift/prometheus-operator/pull/141 get resolved
```
# oc -n openshift-monitoring exec -c prometheus-operator prometheus-operator-787f886bc9-njbk8 -- operator --version
prometheus-operator, version 0.51.2 (branch: master, revision: e2f1f5dde)
  build user:       root
  build date:       20211027-14:17:36
  go version:       go1.16.6
  platform:         linux/amd64

# oc -n openshift-monitoring get deploy prometheus-operator -oyaml
...
  labels:
    app.kubernetes.io/component: controller
    app.kubernetes.io/managed-by: cluster-monitoring-operator
    app.kubernetes.io/name: prometheus-operator
    app.kubernetes.io/part-of: openshift-monitoring
    app.kubernetes.io/version: 0.52.0
```

**Comment by raptorsun** [comment]
> @raptorsun prometheus-operator version is still 0.51.2, although the resource labels is 0.52.0, waiting for [openshift/prometheus-operator#141](https://github.com/openshift/prometheus-operator/pull/141) get resolved
> 
> ```
> # oc -n openshift-monitoring exec -c prometheus-operator prometheus-operator-787f886bc9-njbk8 -- operator --version
> prometheus-operator, version 0.51.2 (branch: master, revision: e2f1f5dde)
>   build user:       root
>   build date:       20211027-14:17:36
>   go version:       go1.16.6
>   platform:         linux/amd64
> 
> # oc -n openshift-monitoring get deploy prometheus-operator -oyaml
> ...
>   labels:
>     app.kubernetes.io/component: controller
>     app.kubernetes.io/managed-by: cluster-monitoring-operator
>     app.kubernetes.io/name: prometheus-operator
>     app.kubernetes.io/part-of: openshift-monitoring
>     app.kubernetes.io/version: 0.52.0
> ```

absolutely, we need to wait for that PR merge. That PR requires a base image [`rhel-8-release-golang-1.17-openshift-4.10` that is a work-in-progress](https://github.com/openshift/ocp-build-data/pull/1143), too.

**Comment by raptorsun** [comment]
waiting for https://github.com/openshift/ocp-build-data/pull/1143

**Comment by simonpasquier** [comment]
We need v0.52.1 now :)

**Comment by raptorsun** [comment]
The problem has been resolved. Now we have the Proemetheus Operator 0.52.1 image 🙂 @juzhao 

/unhold

```➜  clusterbot k exec -it prometheus-operator-767f45bf9-c486z -n openshift-monitoring  -- operator --version
prometheus-operator, version 0.52.1 (branch: master, revision: e6904073d)
  build user:       root
  build date:       20211118-10:14:06
  go version:       go1.17.2
  platform:         linux/amd64```

File: go.mod [hunk 1]
```diff
@@ -6,23 +6,23 @@ require (
...
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus-operator/prometheus-operator v0.50.0
-	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.50.0
-	github.com/prometheus-operator/prometheus-operator/pkg/client v0.50.0
+	github.com/prometheus-operator/prometheus-operator v0.52.1
+	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.52.1
+	github.com/prometheus-operator/prometheus-operator/pkg/client v0.52.1
 	github.com/prometheus/client_golang v1.11.0
-	github.com/prometheus/prometheus v1.8.2-0.20210701133801-b0944590a1c9 // v1.8.2 is misleading as Prometheus does not have v2 module. This is pointing to v2.26.1, the same as in prometheus-operator v0.49.0
+	github.com/prometheus/prometheus v1.8.2-0.20211005150130-f29caccc4255 // v1.8.2 is misleading as Prometheus does not have v2 module. This is pointing to v2.26.1, the same as in prometheus-operator v0.49.0
```
Comment (simonpasquier) [inline go.mod:18]: can you update the comment?

Comment (raptorsun) [reply go.mod]: updated :)

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Comment by simonpasquier** [comment]
/label docs-approved
/label px-approved
/label qe-approved
no functional change, it's a consequence of https://github.com/openshift/prometheus-operator/pull/141

**Pull request merged**